File: GuessNumber/app/src/main/java/com/android/guessnumber/MainActivity.java
package com.android.guessnumber;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private static final String LOG_TAG =
            MainActivity.class.getSimpleName();

    private TextView tvInfo;
    private EditText etInput;
    private Button bControl;
    private int number;
    private boolean isOver;
    AlertDialog.Builder builder;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvInfo = findViewById(R.id.textView);
        etInput = findViewById(R.id.editText);
        bControl = findViewById(R.id.button);
        builder = new AlertDialog.Builder(MainActivity.this);

        newGame();
    }

    private void newGame()
    {
        Random r = new Random();
        int low = 1;
        int high = 10; // 200, 10 to test
        number = r.nextInt(high - low) + low;
        isOver = false;
    }

    public void enterButton(View view) {
        if (etInput.getText().toString().equals(""))
        {
            Log.i(LOG_TAG, String.format("Empty input"));

            builder.setTitle(R.string.error)
                    .setMessage(R.string.empty)
                    .setCancelable(false)
                    .setNegativeButton("OK",
                            new DialogInterface.OnClickListener() {
                                public void onClick(DialogInterface dialog, int id) {
                                    dialog.cancel();
                                }
                            });
            AlertDialog alert = builder.create();
            alert.show();
        }
        else {

            try {
                int temp = Integer.parseInt(etInput.getText().toString());

                if (temp < 1 || temp > 10) {
                    builder.setTitle(R.string.error)
                            .setMessage(R.string.value_borders)
                            .setCancelable(false)
                            .setNegativeButton("OK",
                                    new DialogInterface.OnClickListener() {
                                        public void onClick(DialogInterface dialog, int id) {
                                            dialog.cancel();
                                        }
                                    });
                    AlertDialog alert = builder.create();
                    alert.show();

                    etInput.setText("");
                } else {
                    if (temp > number) {
                        tvInfo.setText(getResources().getString(R.string.ahead));
                    } else if (temp < number) {
                        tvInfo.setText(getResources().getString(R.string.behind));
                    } else if (temp == number) {
                        tvInfo.setText(getResources().getString(R.string.hit));
                        isOver = true;
                        newGame();
                        etInput.setText("");
                    }
                }
            } catch (NumberFormatException e) {
                builder.setTitle(R.string.error)
                        .setMessage(R.string.value_not_int)
                        .setCancelable(false)
                        .setNegativeButton("OK",
                                new DialogInterface.OnClickListener() {
                                    public void onClick(DialogInterface dialog, int id) {
                                        dialog.cancel();
                                    }
                                });
                AlertDialog alert = builder.create();
                alert.show();

                etInput.setText("");
            }
        }

    }
}

File: Calculator/app/src/main/java/com/android/calculator/MainActivity.kt
package com.android.calculator

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.appcompat.app.AlertDialog
import kotlinx.android.synthetic.main.activity_main.*
import java.util.ArrayList

class MainActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener{

    var listOfItemsFrom = arrayListOf("", "")
    var listOfItemsTo = arrayListOf("", "")

    override fun onCreate(savedInstanceState: Bundle?) {

        listOfItemsFrom = resources.getStringArray(R.array.listOfItemsFrom).toCollection(ArrayList<String>())
        listOfItemsTo = resources.getStringArray(R.array.listOfItemsTo).toCollection(ArrayList<String>())

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Create an ArrayAdapter using a simple spinner layout and languages array
        val aa1 = ArrayAdapter(this, android.R.layout.simple_spinner_item, listOfItemsFrom)
        // Set layout to use when the list of choices appear
        aa1.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        // Set Adapter to Spinner
        spinner_from!!.adapter = aa1

        val aa2 = ArrayAdapter(this, android.R.layout.simple_spinner_item, listOfItemsTo)
        aa2.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner_to!!.adapter = aa2
    }



    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        if(parent.id == R.id.spinner_from)
        {
            from.text = listOfItemsFrom[position]
            editText_from.isEnabled = true
        }
        else if(parent.id == R.id.spinner_to)
        {
            to.text = listOfItemsTo[position]
        }
    }

    override fun onNothingSelected(parent: AdapterView<*>) {
        if(parent.id == R.id.spinner_from)
        {
            from.setText(R.string.from)
            editText_from.isEnabled = false
        }
        else if(parent.id == R.id.spinner_to)
        {
            to.setText(R.string.to)
        }
    }


    private val positiveButtonClick = { dialog: DialogInterface, which: Int ->
    }

    private fun basicAlert(view: View){

        val builder = AlertDialog.Builder(this)

        with(builder)
        {
            setTitle(R.string.error)
            setMessage(R.string.wrong_input)
            setPositiveButton("OK", DialogInterface.OnClickListener(function = positiveButtonClick))
            show()
        }


    }

    fun calcButton(view: View) {
        try {
            var fromVal:Double = editText_from.text.toString().toDouble()
            if (fromVal >= 0) {
                if (spinner_from.selectedItem.equals(listOfItemsFrom[1]))
                    fromVal /= 1000000.0

                if (spinner_to.selectedItem.equals(listOfItemsTo[0])) {
                    editText_to.text = (fromVal * 10763910.0).toString()
                } else {
                    editText_to.text = (fromVal * 247.1).toString()
                }
            }
            else
            {
                basicAlert(view)
            }
        } catch (e: NumberFormatException) {
            basicAlert(view)
        }

    }


}
